presto-main/src/main/java/com/facebook/presto/Session.java

```diff
@@ -358,11 +364,22 @@ public ClientSession toClientSession(URI server, boolean debug, Duration clientR
                 timeZoneKey.getId(),
                 locale,
                 properties.build(),
+                encodedStatements.build(),
                 transactionId.map(TransactionId::toString).orElse(null),
                 debug,
                 clientRequestTimeout);
     }
 
+    private static String utf8Encode(String string)
```

Comment:
The name `urlEncode` is better than `utf8Encode` since this is doing URL encoding (using UTF-8, which is standard). The name `utf8Encode` is ambiguous since it could be doing any type of encoding.